JASPER-32 : add the possibility to set json data source in parameters

anasseaharrar opened the pull request:
Replacing JRBeanCollectionDataSource datatype with JRDataSource to permit other jasper datasources to be used
Adding jasperreports-functions library to permit users to use jasper integrated function on the template.
Fix update_db_jasper_2.0.1-2.0.2.sql
Fix manage_jasperreport.html template column name

anasseaharrar committed:
JASPER-32 : add the possibility to set json data source in parameters

Replacing JRBeanCollectionDataSource datatype with JRDataSource to permit other jasper datasources to be used
Adding jasperreports-functions library to permit users to use jasper integrated function on the template.
Fix update_db_jasper_2.0.1-2.0.2.sql
Fix manage_jasperreport.html template column name

jonenst commented:
Thanks for the PR !
Can you please change the following ?
- split the different logical changes in different commits
- create individual JIRA issues for the 4 separate changes
- reorder the new overloaded methods so that each method is next to its new counterpart
- add a comment saying that we override the methods instead of replacing them to ensure binary compatibility (it should work, but can you test it please ? ie test that the new version of this plugin works with old releases of users of this plugin)
- don't change the .gitignore

thanks in advance !

anasseaharrar committed:
JASPER-32 : add the possibility to set json data source in parameters

Revert changes that don't concern the JIRA JASPER-32
Reordering methods
Comment on overrided method

anasseaharrar committed:
Lutece Formatter & update licences : mvn formatter:format license:format

anasseaharrar commented:
Tested on an application that use the DownloadFileServlet Servlet.
This Servlet call one of the overloaded method. 
Tests are Ok for the method call but this new version change one column name in jasper table.
Test done from plugin-jasper-2.0.1 to this version.

jonenst commented:
Merged, Thanks !
(I squashed the first 2 commits)
